obspy/io/gcf/core.py

```diff
@@ -33,68 +142,584 @@ def merge_gcf_stream(st):
                     break
         else:
             traces.append(tr)
-    return Stream(traces=traces)
+    return Stream(traces=traces) 
 
 
 def _is_gcf(filename):
-    """
-    Checks whether a file is GCF or not.
+   """
+   Checks whether a file is GCF or not.
 
-    :type filename: str
-    :param filename: GCF file to be checked.
-    :rtype: bool
-    :return: ``True`` if a GCF file.
-    """
-    try:
-        with open(filename, 'rb') as f:
-            libgcf.is_gcf(f)
-    except Exception:
-        return False
-    return True
+   :type filename: str
+   :param filename: path to GCF file to be checked.
+   :rtype: bool
+   :return: True if a object pointed to by path is a GCF file.
+   """
+   is_gcf = True
+   if not os.path.isfile(filename) or os.path.getsize(filename)%1024:
+      # File either does not point at a file object or file is not of proper size
+      is_gcf = False
+   else:
+      # Load shared library
+      try:
+         gcf_io = _load_cdll("gcf")
+      except Exception as e:
+         print(str(e))
+      else:
+         # declare function argument and return types
+         gcf_io.read_gcf.argtypes = [ctypes.c_char_p, ctypes.POINTER(_GcfFile), ctypes.c_int]
+         gcf_io.read_gcf.restype = ctypes.c_int
+         gcf_io.free_GcfFile.argtypes = [ctypes.POINTER(_GcfFile)]
+         gcf_io.free_GcfFile.restype = None
 
+         # Decode first block
+         obj = _GcfFile()
+         b_filename = filename.encode('utf-8')
+         ret = gcf_io.read_gcf(b_filename, obj, 3)
+         if ret or (obj.n_errHead and obj.seg[0].err not in (10, 11, 21)) or obj.n_errData: 
+            is_gcf = False
+            
+         # release allocated memory
+         gcf_io.free_GcfFile(obj)
+   return is_gcf
 
-def _read_gcf(filename, headonly=False, **kwargs):  # @UnusedVariable
-    """
-    Reads a GCF file and returns a Stream object.
 
-    only GCF files containing data records are supported.
-
-    .. warning::
+def _read_gcf(filename, networkcode='', stationcode='', locationcode='', bandcode="H", instrumentcode="H", channel_prefix=None, 
```

Comment:
in most instances we just have `network` where it's intuitive that a network code is meant and not like a Network object, so i'd vote for just `network`, `station`, etc (`bandcode` and `instrumentcode` are fine, it's just about those NSLC / SEED ID items)

Reply:
Also, you could keep `headonly` at second position. Not that important, since private routines can have breaking changes without notice (and its unlikely that people would use `headonly` as a positional argument rather than a kwarg), but kind of like a low hanging fruit to basically stay compatible with old call.

Reply:
Sure will fix